src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenReadOnlySpanConstructionTest.cs

```diff
@@ -2685,5 +2673,890 @@ public static int M()
+        {
+            var src = $$"""
+var values = C.M();
+System.Console.Write($"{values.Length} {values[0] is null} {values[1] is null} {values[2] is null}");
+
+public class C
+{
+    public static System.ReadOnlySpan<{{type}}> M()
+        => new {{type}}[] { null, null, null };
+
+    public static System.ReadOnlySpan<{{type}}> M2()
+        => new {{type}}[] { null, null, null };
+}
+""";
+            var compilation = CreateCompilationWithMscorlibAndSpan(src);
+            var verifier = CompileAndVerify(compilation, expectedOutput: "3 True True True", verify: Verification.Skipped);
+            var expectedIL = $$"""
+{
+  // Code size       27 (0x1b)
+  .maxstack  2
+  IL_0000:  ldsfld     "{{type}}[] <PrivateImplementationDetails>.709E80C88487A2411E1EE4DFB9F22A861492D20C4765150C0C794ABD70F8147C_B{{typeCode}}"
+  IL_0005:  dup
+  IL_0006:  brtrue.s   IL_0015
+  IL_0008:  pop
+  IL_0009:  ldc.i4.3
+  IL_000a:  newarr     "{{type}}"
+  IL_000f:  dup
+  IL_0010:  stsfld     "{{type}}[] <PrivateImplementationDetails>.709E80C88487A2411E1EE4DFB9F22A861492D20C4765150C0C794ABD70F8147C_B{{typeCode}}"
+  IL_0015:  newobj     "System.ReadOnlySpan<{{type}}>..ctor({{type}}[])"
+  IL_001a:  ret
+}
+""";
+            verifier.VerifyIL("C.M", expectedIL);
+            verifier.VerifyIL("C.M2", expectedIL);
+
+            compilation = CreateCompilationWithMscorlibAndSpan(new[] { RuntimeHelpersCreateSpan, src });
+            verifier = CompileAndVerify(compilation, verify: Verification.Skipped);
+            verifier.VerifyIL("C.M", expectedIL);
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69472")]
+        public void ReadOnlySpanFromArrayOfConstants_Null_MultipleTypes()
+        {
+            var src = $$"""
+Assert(C.MString());
+Assert(C.MString2());
+Assert(C.MObject());
+Assert(C.MObject2());
+Assert(C.MC());
+Assert(C.MC2());
+
+System.Console.Write("ran");
+
+static void Assert<T>(System.ReadOnlySpan<T> values)
+{
+    if (values.Length != 3) throw null;
+    if (values[0] is not null) throw null;
+    if (values[1] is not null) throw null;
+    if (values[2] is not null) throw null;
+}
+
+public class C
+{
+    public static System.ReadOnlySpan<string> MString() => new string[] { null, null, null };
+    public static System.ReadOnlySpan<string> MString2() => new string[] { null, null, null };
+
+    public static System.ReadOnlySpan<object> MObject() => new object[] { null, null, null };
+    public static System.ReadOnlySpan<object> MObject2() => new object[] { null, null, null };
+
+    public static System.ReadOnlySpan<C> MC() => new C[] { null, null, null };
+    public static System.ReadOnlySpan<C> MC2() => new C[] { null, null, null };
+}
+""";
+            var compilation = CreateCompilationWithMscorlibAndSpan(src);
+            var verifier = CompileAndVerify(compilation, expectedOutput: "ran", verify: Verification.Skipped);
```

Comment:
>verify: Verification.Skipped [](http://example.com/codeflow?start=80&length=28)

Why is verification skipped for this scenario?

Reply:
```
PEVerify failed for assembly 'C:\Users\jcouv\AppData\Local\Temp\RoslynTests':
[ : C::MString][mdToken=0x6000006][offset 0x00000010] Cannot change initonly field outside its .ctor.
[ : C::MObject][mdToken=0x6000007][offset 0x00000010] Cannot change initonly field outside its .ctor.
[ : C::MC][mdToken=0x6000008][offset 0x00000010] Cannot change initonly field outside its .ctor.
[ : C::MC][mdToken=0x6000008][offset 0x00000015][found ref array mdarray 'System.Object[]'][expected ref array mdarray 'C[]'] Unexpected array type on the stack.
```
```
System.Exception : IL Verify failed unexpectedly:
[MString]: Cannot change initonly field outside its .ctor. { Offset = 0x10 }
[MString]: Return type is ByRef, TypedReference, ArgHandle, or ArgIterator. { Offset = 0x1a }
[MObject]: Cannot change initonly field outside its .ctor. { Offset = 0x10 }
[MObject]: Return type is ByRef, TypedReference, ArgHandle, or ArgIterator. { Offset = 0x1a }
[MC]: Cannot change initonly field outside its .ctor. { Offset = 0x10 }
[MC]: Unexpected type on the stack. { Offset = 0x15, Found = ref 'object[]', Expected = ref '[02d07b78-95a0-409e-bff4-1cbc39fcff23]C[]' }
[MC]: Return type is ByRef, TypedReference, ArgHandle, or ArgIterator. { Offset = 0x1a }
```

Reply:
> [ : C::MString][mdToken=0x6000006][offset 0x00000010] Cannot change initonly field outside its .ctor.

This looks like a real problem. It looks like we shouldn't be marking the field `readonly`.

> [ : C::MC][mdToken=0x6000008][offset 0x00000015][found ref array mdarray 'System.Object[]'][expected ref array mdarray 'C[]'] Unexpected array type on the stack.

This is concerning as well.
1. At the very least, it looks like we should pay attention to ```IsPeVerifyCompatEnabled``` after all (and test that as well).
2. Are we confident that we are not going to run in trouble trying to consume the span. Could that trigger some hard runtime type check failure in some scenarios, For example, when we try to store an element reference in a `ref readonly` local, etc. ? 

It looks like we should verify expected verification failure reasons in the tests targeting scenarios affected by this PR.

Reply:
Made the field writable and removed arbitrary reference type scenario, so there's no PEVerify regression (so didn't add any tests targeting `IsPeVerifyCompatEnabled` case). 
I added some usage of the ROS with a `ref readonly` local.